### PR TITLE
[Refactor] Simplify QueryService

### PR DESCRIPTION
**Pull request by mproch**
Main aim is to use NodeCompiler when preparing service to invoke, so that logic (especially WRT dynamic, eager services) can be reused

### CODE REVIEW

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/util/service/query/ServiceQuery.scala [hunk 1]
```diff
@@ -17,61 +28,74 @@ class ServiceQuery(modelData: ModelData) {
...
-      collector.cleanResults()
+      val collector = QueryServiceInvocationCollector(serviceName)
+        .enable(TestRunId(UUID.randomUUID().toString))
+
+      val variablesPreparer = GlobalVariablesPreparer(definitions.expressionConfig)
+      val validationContext = variablesPreparer.emptyValidationContext(metaData)
+
+      val compiled = compiler.compileService(ServiceRef(serviceName, params), validationContext, None, Some(_ => collector))(NodeId(""), metaData)
+      compiled.compiledObject.map { service =>
+          service.invoke(ctx, ExpressionEvaluator.unOptimizedEvaluator(variablesPreparer))._2.map(QueryResult(_, collector.getResults))
```
Comment (jedrz) [inline engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/util/service/query/ServiceQuery.scala:62]: Shouldn't `evaluator` be used here?

File: ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ServiceRoutesSpec.scala [hunk 2]
```diff
@@ -70,23 +69,24 @@ class ServiceRoutesSpec extends FunSuite with Matchers with ScalatestRouteTest w
       """.stripMargin)
     Post("/service/streaming/enricher", entity) ~> serviceRoutes.securedRoute ~> check {
       status shouldEqual StatusCodes.InternalServerError
-      entityAs[JsonThrowable].message shouldEqual Some("NonEmptyList(ExpressionParseError(EL1041E: After parsing a valid expression, there is still more data in the expression: 'spell',defaultNodeId,Some(param),not valid spell expression))")
-      entityAs[JsonThrowable].className shouldEqual classOf[ParametersCompilationException].getCanonicalName
+      entityAs[JsonThrowable].message shouldEqual Some("MissingParameters(Set(tariffType),)")
```
Comment (jedrz) [inline ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ServiceRoutesSpec.scala:72]: What about adding tariffType parameter to the invocation so the error should remain more or less the same?

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/util/service/query/ServiceQuery.scala (hunk 1 above)
Comment (mproch) [reply engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/util/service/query/ServiceQuery.scala]: :+1: done

File: ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ServiceRoutesSpec.scala (hunk 2 above)
Comment (mproch) [reply ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ServiceRoutesSpec.scala]: :+1: done

**Review by jedrz — APPROVED** [review]

**Pull request merged**